packages/jsii/lib/assembler.ts

```diff
@@ -892,6 +894,45 @@ export class Assembler implements Emitter {
     return [jsiiType];
   }
 
+  private _validateHeritageClauses(clauses?: ts.NodeArray<ts.HeritageClause>) {
+    if (clauses == null || clauses.length === 0) {
+      // Nothing to do.
+      return;
+    }
+    for (const clause of clauses) {
+      for (const node of clause.types) {
+        const parentType = this._typeChecker.getTypeAtLocation(node);
+        // For some reason, we cannot trust parentType.isClassOrInterface()
```

Comment:
Expand a bit with an example of when it has lost your trust?

Reply:
Basically, imported types respond `false` to all `is*` methods. :(